File: src/main/java/com/example/rocko/InvertBinaryTree.java
package com.example.rocko;


import com.example.rocko.lib.TreeNode;

/**
 * Created by Rocko on 2016/12/2.
 */
public class InvertBinaryTree {

    public TreeNode invertTree(TreeNode root) {
        if(root == null){
            return null;
        }

        TreeNode tmp = root.left;
        root.left = root.right;
        root.right = tmp;

        invertTree(root.left);
        invertTree(root.right);

        return root;
    }
}

File: src/InvertBinaryTree.php
<?php

namespace Rocko;

class InvertBinaryTree
{
    public function invertTree($originTree)
    {
        if (is_null($originTree)) {
            return null;
        }
        
        $tmp = $originTree->left;
        $originTree->left = $originTree->right;
        $originTree->right = $tmp;
        
        $this->invertTree($originTree->left);
        $this->invertTree($originTree->right);
        
        return $originTree;
        
    }
}